crates/provider/src/evm/rpc/values_provider.rs

```diff
@@ -0,0 +1,246 @@
+//! This module provides a `TrieValueProvider` struct that fetches trie values from an Ethereum node using JSON-RPC.
+
+use alloy_primitives::B256;
+use anyhow::{anyhow, Result};
+use hdp_primitives::block::{account::AccountProofFromRpc, header::BlockHeaderFromRpc};
+use reqwest::{header, Client};
+use serde_json::{from_value, json, Value};
+use std::str::FromStr;
+
+use crate::key::{AccountProviderKey, HeaderProviderKey, StorageProviderKey};
+
+pub struct TrieValueProvider {
+    client: Client,
+    pub url: &'static str,
+}
+
+impl TrieValueProvider {
+    pub fn new(rpc_url: &'static str) -> Self {
+        Self {
+            client: Client::new(),
+            url: rpc_url,
+        }
+    }
+
+    pub async fn get_blocks_by_keys(
+        &self,
+        provider_keys: &[HeaderProviderKey],
+    ) -> Result<Vec<BlockHeaderFromRpc>> {
+        let mut blocks = Vec::new();
+        for provider_key in provider_keys {
+            let block = self.get_block_by_number(provider_key).await?;
+            blocks.push(block);
+        }
+        Ok(blocks)
+    }
+
+    async fn get_block_by_number(
+        &self,
+        provider_key: &HeaderProviderKey,
+    ) -> Result<BlockHeaderFromRpc> {
```

Comment:
same here cant alloy types be used here as it is standard rpc call?

Reply:
same